src/widget/mod.rs

```diff
@@ -1,15 +1,21 @@
 pub mod container;
 pub mod text;
 
+use alloc::boxed::Box;
 use embedded_graphics::prelude::*;
 
-pub trait Widget<Color> {
-    fn draw<Display, Error>(
```

Comment:
Traits with type parameters on functions aren't object-safe, so can't be boxed, so I had to hoist `Display` up to the trait level. That's kind of unfortunate, since it adds quite a bit of `where Display: ...` boilerplate all over the place.